### PR TITLE
Support binary columns with SQL Server FREETEXT/CONTAINS

### LINKED ISSUE #23921: SqlServer: Allow FreeText and Contains function to take non-string parameters
I have the following entity with a property `Content` of type `byte[]`:

```cs
public class Post
{
  [Column(TypeName = "varbinary(max)")]
  [Required]
  public byte[] Content { get; set; }
  
  public string ContentType { get; set; }
}
```

and the content type in model builder:
```
modelBuilder.Entity<Post>()
    .Property(p => p.ContentType)
    .HasComputedColumnSql("'.html'");
```

The Migration:

```cs
migrationBuilder.Sql("CREATE FULLTEXT CATALOG ft_catalog AS DEFAULT", true);
migrationBuilder.Sql(@"CREATE FULLTEXT INDEX ON dbo.Posts([Content] TYPE COLUMN ContentType) KEY INDEX PK_Posts", true);
```

`EF.Functions.FreeText` takes a string `propertyReference` and a string `freeText`.

Here, I passed the Column "Content" as a string I got `The expression passed to the 'propertyReference' parameter of the 'FreeText' method is not a valid reference to a property. The expression should represent a reference to a full-text indexed property on the object referenced in the from clause`.

```cs
IQueryable<DataSets.Post> query = db.Posts;

query = query.Where(k => EF.Functions.FreeText("Content", model.keyword)); 
```

Then, I looked around and found `EF.Property` and used it on `propertyReference` param, Then I got `Failed to convert parameter value from a String to a Byte[]`.
```cs
query = query.Where(k => EF.Functions.FreeText(EF.Property<string>(k, "Content"), model.keyword));
```

The following SQL query works perfectly against the database:

```sql
select ID,Title,Description from Posts where FREETEXT(Content,'wolf')
```
Is there something special for byte[] to be done without having that error?

### PR DESCRIPTION
**Pull request by roji**
Closes #23921

### CODE REVIEW

File: src/EFCore.SqlServer/Extensions/SqlServerDbFunctionsExtensions.cs [hunk 1]
```diff
@@ -554,10 +554,10 @@ public static class SqlServerDbFunctionsExtensions
             => throw new InvalidOperationException(CoreStrings.FunctionOnClient(nameof(DateDiffMillisecond)));
 
         /// <summary>
-        ///     Counts the number of millisecond boundaries crossed between the startDate and endDate.
-        ///     Corresponds to SQL Server's DATEDIFF(millisecond, @startDate, @endDate).
+        ///     Counts the number of millisecond boundaries crossed between the <paramref name="startDate" /> and
+        ///     <paramref name="endDate" />. Corresponds to SQL Server's <c>DATEDIFF(millisecond, @startDate, @endDate)</c>.
         /// </summary>
-        /// <param name="_">The DbFunctions instance.</param>
+        /// <param name="_">The <see cref="DbFunctions" /> instance.</param>
```
Comment (smitpatel) [inline src/EFCore.SqlServer/Extensions/SqlServerDbFunctionsExtensions.cs:560]: Now that you have decided to update this
also DbFunctionsExtensions/Sqlite version/Relational version please 😉

**Comment by roji** [comment]
I know adding stuff to Northwind would end badly. @maumar @smitpatel suggestions on where this test should live? GearsOfWar?

**Comment by smitpatel** [comment]
I am not happy yet. Following 2 files needs similar update for docs.
https://github.com/dotnet/efcore/blob/FreeTheText/src/EFCore/DbFunctionsExtensions.cs
https://github.com/dotnet/efcore/blob/FreeTheText/src/EFCore.Relational/Extensions/RelationalDbFunctionsExtensions.cs

For testbase, @maumar is the boss!

**Comment by roji** [comment]
@smitpatel next task!

File: test/EFCore.SqlServer.FunctionalTests/Query/GearsOfWarQuerySqlServerFixture.cs [hunk 2]
```diff
@@ -16,6 +16,47 @@ protected override void OnModelCreating(ModelBuilder modelBuilder, DbContext con
...
+BEGIN
+    IF EXISTS (SELECT 1 FROM sys.fulltext_catalogs WHERE [name] = 'GearsOfWar_FTC')
+    BEGIN
+        DROP FULLTEXT CATALOG GearsOfWar_FTC;
+    END
+
+    CREATE FULLTEXT CATALOG GearsOfWar_FTC AS DEFAULT;
+    CREATE FULLTEXT INDEX ON Missions (BriefingDocument TYPE COLUMN BriefingDocumentFileExtension) KEY INDEX PK_Missions;
+
+    WAITFOR DELAY '00:00:03';
```
Comment (roji) [inline test/EFCore.SqlServer.FunctionalTests/Query/GearsOfWarQuerySqlServerFixture.cs:58]: Yes. Unfortunately there is some asynchronous process here, so if we run the tests too quickly after creating the full text catalog/index, we get back empty resullts (not an error, which is what you get when the catalog/index aren't created at all). :rage: :rage: :rage:

**Review by smitpatel — COMMENTED** [review]
LGTM. @maumar to sign-off for test changes.

**Review by maumar — APPROVED** [review]

**Pull request merged**